### PR TITLE
adding support for no_std executor

### PR DESCRIPTION
**Pull request by richardanaya**
Hey all, let me know what you think!

### CODE REVIEW

File: src/executors.rs [hunk 1]
```diff
@@ -132,3 +132,40 @@ impl AgnostikExecutor for BastionExecutor {
...
+}
+
+#[cfg(feature = "runtime_nostd")]
+impl AgnostikExecutor for NoStdExecutor {
+    fn spawn<F, T>(&self, _future: F) -> JoinHandle<T>
+    where
+        F: Future<Output = T> + Send + 'static,
+        T: Send + 'static,
+    {
+        panic!("no threads on no_std environments")
```
Comment (Nemo157) [inline src/executors.rs:164]: Threads aren't necessary for spawning tasks, it's possible to have fully capable core-only executors, they're just going to necessarily be much more device specific than any using `std`.

Comment (richardanaya) [reply src/executors.rs:164]: awesome, I thought someone might say something about this, I'm curious what people think should be done about these cases for `no_std` , a better message? no_std multithreading is a bit out of my knowledge base.

**Comment by o0Ignition0o** [comment]
Let's go for an other CI run to see how it goes.

**Comment by Stupremee** [comment]
Hey @richardanaya 
Any progress with the no_std executor? 
Because I would like to have a no_std executor in the next release of agnostik

**Comment by richardanaya** [comment]
@Stupremee not having good luck:

i'm getting a new error about from_generator not existing after doing a pull from master, and still getting an error on nightly ( after rustup update nightly ).

Also, I could still use an answer for what people think I should do for no_std multithreading.

**Comment by o0Ignition0o** [comment]
Hey there :) Is there any way I can help ? I tried checking out the code, and rustc seemed to complain about phantomdata not implementing derefmut. I tried to look for a joinhandle of some sort but couldnt find one :)

Lemme know if theres anything i can do to help, i ll be available tomorrow (CET) :)

**Comment by o0Ignition0o** [comment]
just ran into it, will try to figure it out :)

```
error[E0433]: failed to resolve: could not find `from_generator` in `future`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0433`.
error: could not compile `agnostik`.
```

It should be fixed somewhere https://github.com/rust-lang/rust/pull/69033

File: src/executors.rs (hunk 1 above)
Comment (Nemo157) [reply src/executors.rs:164]: :man_shrugging: I would say that `no_std` users should likely just supply their own, it's unlikely that there's a decent one-size-fits-all approach possible.

Comment (richardanaya) [reply src/executors.rs:164]: Yah, I agree, i'm a web assembly dev and I don't even really have an option in this respect :)

**Comment by richardanaya** [comment]
what really confused me about the from_generator issue is what is causing it. My library `executor` compiles on stable just fine. I could have sworn it only came from using `async { }` blocks, but I didn't see any blocks that would be used by `runtime_nostd` agnostic, its like something sneaky is leaking in.

**Comment by Stupremee** [comment]
I thought about this the last few days and came to a conclusion.
First, I'm sure we don't need your executor and just implement ourselves.
The methods should look like this:

- **spawn & spawn_blocking:** Should be noop methods
- **block_on:** call `Future::poll` until the future is ready

If you want you can add agnostik support to your executor library as soon as #4 is implemented

**Comment by richardanaya** [comment]
Cool, closing this PR.

**Comment by cosmikwolf** [comment]
And 3.5 years later, no_std is still not implemented, and #4 is still not implemented. 

maybe the lesson here is to take help when someone is offering for free. that way we would have a rust ecosystem with an abstraction layer that is compatible with no_std.

**Pull request closed without merging**